packages/calcite-components/src/components/button/button.scss

```diff
@@ -1,64 +1,144 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-button-background-color: Specifies the component's background color.
+ * @prop --calcite-button-border-color: Specifies the component's border color.
+ * @prop --calcite-button-corner-radius: Specifies the component's corner radius.
+ * @prop --calcite-button-corner-radius-start-start: Specifies the component's top-left corner radius. will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-start-end: Specifies the component's top-right corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-end-start: Specifies the component's bottom-left corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-corner-radius-end-end: Specifies the component's bottom-right corner radius. Will fallback to --calcite-button-corner-radius.
+ * @prop --calcite-button-icon-color: Specifies the color of an icon in the component.
+ * @prop --calcite-button-loader-color: Specifies the component's loader color.
+ * @prop --calcite-button-shadow-color: Specifies the component's shadow.
+ * @prop --calcite-button-text-color: Specifies the component's text color.
+ */
+
 :host {
   @apply inline-block w-auto align-middle;
-}
+  border-start-start-radius: var(
+    --calcite-button-corner-radius-start-start,
+    var(--calcite-button-corner-radius, var(--calcite-internal-button-corner-radius, var(--calcite-corner-radius)))
+  );
```

Comment:
If there is a public `--calcite-button-corner-radius`, why do we have an internal version of it `var(--calcite-internal-button-corner-radius`? This came up in action PR, so I just wanted to clarify.

Reply:
This allows the public tokens to still be changeable by users while allowing other components to make adjustments to the default

Reply:
this got messy so I ended but simplifying it back to just a border-radius on the :host